packages/actor-init-query/lib/cli/CliArgsHandlerBase.ts

```diff
@@ -145,6 +147,45 @@ export class CliArgsHandlerBase implements ICliArgsHandler {
       .help(false);
   }
 
+  public static URIfy (source: string): string {
+    // In order to make running comunica on local files a more pleasant experience
+    // file system paths are supported as source arguments,
+    // being transformed into URIs under the hood.
+    // As requested in https://github.com/comunica/comunica/issues/1044
+
+    function isURI (str: string): boolean {
+      let URIRegex: RegExp = /^\w[\w\d\+\.\-]*:/;
+      // However, it also matches Windows drive syntax, such as C:
+
+      // In order to distinguish between a drive file path and a URI
+      // we could check with a drive file path regex:
+      let windowsDriveFilePathRegex: RegExp = /^\w:/;
+      let isWindowsDrivePath: boolean = str.match(windowsDriveFilePathRegex) !== null;
+
+      // Also comunica supports tagged URIs
+      let taggedURIRegExp: RegExp = /^\w+@\w[\w\d\+\.\-]*:/;
+      let isTaggedURI: boolean = str.match(taggedURIRegExp) !== null;
+
+      let isURI: boolean = isTaggedURI || !isWindowsDrivePath && str.match(URIRegex) !== null;
+      return isURI;
+    }
+
+    // Based on:
+    // https://github.com/comunica/comunica/blob/715b8acc122d28d09cbe28c02a1c7b8a7b82d024/packages/actor-dereference-file/test/ActorDereferenceFile-test.ts#L10
+    function fileUri(str: string): string {
+        let pathName = path.resolve(str).replace(/\\/ug, '/');
+
+        // Windows drive letter must be prefixed with a slash
+        if (!pathName.startsWith('/')) {
+            pathName = `/${pathName}`;
+        }
+
+        return encodeURI(`file://${pathName}`);
+    }
+
+    return isURI(source) ? source : fileUri(source);
```

Comment:
I think the logic above can be simplified quite a lot.
I guess we should only be checking for the presence of a scheme (`http:`, `https://`, ...), and if it is not present, fallback to the `file://` scheme.

Reply:
Yeah, that would be great.
I've made a wrong assumption that values such as `C:` are valid URIs.

Looks like all we need is check whether it's a (tagged) URI, and if not - fallback to deriving a file URI for it, as you suggest. Thanks for the tip.